vignettes_src/cuda.Rmd

```diff
@@ -0,0 +1,326 @@
+---
+title: "dust: Running models on GPUs with CUDA"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{dust: Running models on GPUs with CUDA}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  error = FALSE,
+  collapse = TRUE,
+  comment = "#>"
+)
+lang_output <- function(x, lang) {
+  cat(c(sprintf("```%s", lang), x, "```"), sep = "\n")
+}
+cc_output <- function(x) lang_output(x, "cc")
+r_output <- function(x) lang_output(x, "r")
+plain_output <- function(x) lang_output(x, "plain")
+```
+
+With the core approach in dust, you can run models in parallel efficiently up to the number of cores your workstation has available.  Getting more than 32 or 64 cores is hard though, and `dust` provides no multi-node parallelism (e.g., [MPI](https://en.wikipedia.org/wiki/Message_Passing_Interface)). Instead, we have developed a system for running `dust` models on GPUs (graphical processing units), specifically NVIDIA GPUs via CUDA (Compute Unified Device Architecture).
+
+This vignette is written in reverse-order, starting with how to run a model on a GPU, before covering how to write a `dust` model that can be run on a GPU. The reason for this is that we do not expect that people will write these directly; instead we expect that most people will use the [`odin.dust`](https://mrc-ide.github.io/odin.dust/) package to generate these interfaces automatically, without having to write as line of C++.
+
+## Principles
+
+* A GPU model can be run on the CPU
+*
+
+## Running a model with GPU support
```

Comment:
Add a section on run requirements? I think these would be:
- CUDA toolkit v10.2 or higher, v11.1 or higher preferred (compile time and run time)
- CUDA capable GPU (run time)
- nvidia drivers (run time). 

Check with `nvidia-smi`